system_baseline/validators.py

```diff
@@ -46,6 +46,29 @@ def check_for_empty_name_values(facts):
             raise FactValidationError("value for %s cannot be empty" % fact["name"])
 
 
+def check_for_invalid_whitespace_name_values(facts):
+    """
+    check if any name or values have invalid whitespace at beginning and end; raises an exception.
+    """
+    for fact in facts:
+        if "values" in fact:
+            check_for_invalid_whitespace_name_values(fact["values"])
+        if "name" in fact and not _check_whitespace(fact["name"]):
+            raise FactValidationError(
+                "fact name cannot have leading or trailing whitespace"
+            )
+        elif "value" in fact and not _check_whitespace(fact["value"]):
+            raise FactValidationError(
+                "value for %s cannot have leading or trailing whitespace" % fact["name"]
+            )
+
+
+def _check_whitespace(input_string):
```

Comment:
if you wanted, you could remove the leading `_` from this name, then call `check_whitespace` from the view instead of rewriting the lstrip/rstrip for `display_name`